### ISSUE TITLE
New release for v2.0.0

### ISSUE THREAD

**Issue by AkshayPall** [ical4j/ical4j#284]
I am working on a project with a critical reliance on ical4j. Currently, we're on v2.2.0 and have been noticed an issue for which I raised a PR and it was merged. We'd like to upgrade however we are currently tight on time and would prefer to minimize the amount of changes required.

To confirm, you don't maintain multiple release versions concurrently, right? In order for me to take advantage of this new fix we're forced to update to 3.0.X, right?

**Comment by benfortuna** [comment]
Hi,

The ical4j v2.x stream is built from a different branch as the current develop branch is v3.x (Java 8 minimum).

If you require v2.x please create a PR for the 2.x-maintenance branch - I'll provide a new release following the merge:

https://github.com/ical4j/ical4j/tree/ical4j-2.x-maintenance

regards,
ben

**Comment by AkshayPall** [comment]
Hi Ben,

Awesome. Thanks for the fast response! I've put up the new PR #285 .

**Comment by benfortuna** [comment]
One thing to note in your tests is how leap year is handled for `FREQ=MONTHLY;BYMONTH=2;INTERVAL=1`.

In your test you assume the next matching date after `20200229` will be `20210228`. This is how the ical4j 2.x branch will handle it, however in ical4j 3.x some recent changes to the implementation mean the next matching date will be `20240229`.

The following text from discussions on behaviour of implicit recurrence rules explains this change:

`when the ‘FREQ’ parameter is set to ‘MONTHLY, but without a ‘BYMONTHDAY’ parameter, the ‘BYMONTHDAY’ selection is inherited from the calendar month of year value from the initial start date;`

Let me know if you think this behaviour will cause problems.

regards,
ben

**Comment by AkshayPall** [comment]
Thanks for making a release so quickly! We've pulled it and everything seems great on our side now.
----
It seems that this change would result in that test case returning a `null` as it would hit the limit of 12 increments. If that is a valid set of recurrence rules, then we could set the increment (`multiplier`) to max at 36 instead, in which case it'd reach the valid date of February 29, 2024.

I'm curious though, this same test passed on the `develop` branch. Was there another reason why I didn't see this behavioural change there? https://github.com/ical4j/ical4j/pull/282/files#diff-a70585d66a7cc664cdb5f0708a985586R852

**Comment by benfortuna** [comment]
Yes, it passed on develop branch because I haven't yet merged the changes to recurrence calculations. Will probably do this pretty soon.

I noticed in fact the test didn't return null, but rather did return the correct 2024 date. The reason is that increment() is only applied in Recur.getDates() but I believe the test is using Recur.getNextDate().

So those tests may not be testing the change as yet, but we can add some more tests specifically for getDates().

**Comment by AkshayPall** [comment]
Oh I see, that clears it up, seems like we will need to add those tests.

Thanks again!
